crates/runner/src/memory/mod.rs

```diff
@@ -45,9 +46,11 @@ pub enum MemoryError {
 /// Memory is addressable by `M31` field elements and stores `QM31` values.
 #[derive(Debug, Clone, Default)]
 pub struct Memory {
-    /// The index of the vector corresponds to the memory address.
-    /// Instructions and data are stored as `QM31` values.
-    pub data: Vec<QM31>,
+    /// Memory is split between local values containing the program and stack, and the heap.
+    /// From the VM point of view, there is no distinction between the two.
+    /// Addresses close to 0 will be appended to the locals vector, and addresses close to 1 << MAX_MEMORY_SIZE_BITS - 1 will be appended to the heap vector.
```

Comment:
Should had the fact that heap is going downards from  MAX_ADDRESS
(replace 1 << MAX_MEMORY_SIZE_BITS - 1  by MAX_ADDRESS)